### PR TITLE
Debloat source code

### PR DESCRIPTION
**Pull request by thatOneArchUser**
no comment just read main.py

### CODE REVIEW

**Comment by thatOneArchUser** [comment]
merge/close/delete whatever not merge is called so i can delete the fork

File: main.py [hunk 1]
```diff
@@ -27,50 +33,31 @@
...
-    shopitem = json.load(f)
+with open('database/currency.json', 'r') as f: currency = json.load(f)
+with open('database/warnings.json', 'r') as f: warnings = json.load(f)
+with open('database/items.json', 'r') as f: items = json.load(f)
+with open('config/shop.json', 'r') as f: shopitem = json.load(f)
 
 #Pre-Initialization Commands
-def timenow(): 
-    datetime.datetime.now().strftime("%H:%M:%S")
+def timenow():  datetime.datetime.now().strftime("%H:%M:%S")
```
Comment (notsniped) [inline main.py:45]: *Remember to remove extra indents later*

**Comment by notsniped** [comment]
Some parts of code look old, I'll give it a look

File: main.py [hunk 2]
```diff
@@ -512,20 +404,15 @@ async def give(ctx:SlashContext, user:discord.User, amount:int):
...
-    if (currency['wallet'][str(user.id)] < 5000):
-        await ctx.reply('They has less than 5000 coins on them. Don\'t waste your time...')
-        return
-    elif (currency['wallet'][str(ctx.author.id)] < 5000):
-        await ctx.reply('You have less than 5k coins in your wallet. Play fair dude.')
-        return
-    if (chance <= 50):
-        x:int = random.randint(5000, currency['wallet'][str(user.id)])
+    if currency['wallet'][str(user.id)] < 5000: return await ctx.reply('They has less than 5000 coins on them. Don\'t waste your time...') 
+    elif currency['wallet'][str(ctx.author.id)] < 5000: return await ctx.reply('You have less than 5k coins in your wallet. Play fair dude.') #what..? wheres logic?? i mean irl you can rob someone without havin any money at all
```
Comment (notsniped) [inline main.py:408]: Technically, robbing someone without having any money is possible, but there also has to be a failure penalty. If it was still 5000 coins from wallet, then the user would be in "debt"

File: main.py [hunk 3]
```diff
@@ -728,45 +556,14 @@ async def hunt(ctx:SlashContext):
...
-        'jellyfish',
-        'shark',
-        'nothing'
-    ]
+    if (items[str(ctx.author.id)]['fishingpole'] == 0): return await ctx.reply('I don\'t think you can fish with your bare hands. Please buy a fishing pole from the shop. ||/buy fishingpole||') #just put yo hands in the water bro **giga chad**
+    loot = ['shrimp', 'fish', 'rare fish', 'exotic fish', 'jellyfish', 'shark', 'nothing']
     choice = random.choice(loot)
-    if (choice == "shrimp"):
-        items[str(ctx.author.id)]['shrimp'] += 1
+    if choice != "nothing":
```
Comment (notsniped) [inline main.py:562]: I think the choices and the keys should be checked again

File: main.py [hunk 4]
```diff
@@ -837,12 +607,7 @@ async def dig(ctx:SlashContext):
     create_option(name='text', description='What do you want to send?', option_type=3, required=True)
   ]
 )
-async def echo(ctx:SlashContext, text:str):
-    try:
-        await ctx.reply("Message echoed!", hidden=True)
-        await ctx.channel.send(text)
-    except Exception as e:
-        await ctx.send(f"An error occured while processing this request. {e}", hidden=True)
+async def echo(ctx:SlashContext, text:str): await ctx.send(text)
```
Comment (notsniped) [inline main.py:610]: I don't think you want to show the name of the person who ran the command...

**Review by notsniped — CHANGES_REQUESTED** [review]
From line 562 - line 566: Please re-check the keys and choices again, I will attempt to fix later.
~~Line 610: Changes will probably be remade for `/echo`~~ resolved

File: main.py (hunk 4 above)
Comment (notsniped) [reply main.py]: Fixed myself :+1:

File: main.py (hunk 3 above)
Comment (notsniped) [reply main.py:562]: Ehh, I fixed it

File: main.py (hunk 2 above)
Comment (thatOneArchUser) [reply main.py:408]: if wallet == 0: return
else: random.randint(round(wallet/10, 0), wallet)

**Comment by thatOneArchUser** [comment]
help what does changes requested mean i dont speak github

**Comment by notsniped** [comment]
Meaning you have to make changes to the code to merge the pull request
However, I have already worked them out

**Review by notsniped — APPROVED** [review]

**Comment by notsniped** [comment]
My debug environment is not working, so I am keeping this pull request on hold.

File: main.py (hunk 2 above)
Comment (notsniped) [reply main.py:409]: I'll try it with this too. Lets see what works

**Comment by notsniped** [comment]
I am probably moving some parts of this pull request to smaller PR's so that I can work on these easily

**Comment by thatOneArchUser** [comment]
what is going on help i dfont speak githb

**Comment by notsniped** [comment]
> what is going on help i dfont speak githb

I just mentioned this pull request in another pull request

**Comment by archisha69** [comment]
nobody speaks github

**Comment by notsniped** [comment]
> brok
> 
> image

***@***.***

**Comment by notsniped** [comment]
**Tasks to be completed:**

- [ ] Merge imports
- [ ] Check database functionality
- [ ] Check events
- [ ] Check changed commands
- [ ] Move code guidelines to separate markdown file

File: main.py [hunk 5]
```diff
@@ -219,22 +158,14 @@ async def unload(ctx:SlashContext, module:str):
 @slash.slash(name='balance', description='Shows your own or another user\'s balance.', options=[create_option(name='user', description='Which user?', option_type=6, required=False)])
```
Comment (notsniped) [inline main.py:158]: Checking from here...

File: main.py [hunk 6]
```diff
@@ -479,8 +380,7 @@ async def scout(ctx:SlashContext):
         currency["wallet"][str(ctx.author.id)] += x
         save()
         await ctx.send(embed=discord.Embed(title='What you found', description=f'You searched your area and found {x} coin(s)!'))
-    else:
-        await ctx.send(embed=discord.Embed(title='What you found', description='Unfortunately no coins for you :('))
+    else: await ctx.send(embed=discord.Embed(title='What you found', description='Unfortunately no coins for you :('))
 
 @slash.slash(
```
Comment (notsniped) [inline main.py:385]: Check from here...

**Comment by thatOneArchUser** [comment]
oh my, i made this pill request more than amonth ago and its still open

**Comment by thatOneArchUser** [comment]
bro wtf i clicked close instead of comment again

**Comment by notsniped** [comment]
*chokes*

**Comment by notsniped** [comment]
> oh my, i made this pill request more than amonth ago and its still open

I'm only now checking and debugging the code XD

**Comment by thatOneArchUser** [comment]
> _chokes_

that doesnt look very professional to me, you are on github m8
:raised_eyebrow:

**Comment by archisha69** [comment]
> > _chokes_
> 
> that doesnt look very professional to me, you are on github m8 raised_eyebrow

o         k    i guess (suddenly cosine when i typed "k")

**Comment by thatOneArchUser** [comment]
> > > _chokes_
> > 
> > 
> > that doesnt look very professional to me, you are on github m8 raised_eyebrow
> 
> o k i guess (suddenly cosine when i typed "k")

kosine

**Comment by notsniped** [comment]
> > _chokes_
> 
> that doesnt look very professional to me, you are on github m8 raised_eyebrow

*unchokes*

File: main.py [hunk 7]
```diff
@@ -571,17 +457,11 @@ async def bankrob(ctx:SlashContext, user:discord.User):
 )
 async def inventory(ctx:SlashContext, user:discord.User = None):
```
Comment (notsniped) [inline main.py:458]: Checking from here...

File: main.py [hunk 8]
```diff
@@ -545,20 +436,15 @@ async def rob(ctx:SlashContext, user:discord.User):
...
-    if (currency['wallet'][str(user.id)] < 10000):
-        await ctx.reply('You really want to risk losing your life to a poor person? (imagine robbing someone with < 10k net worth)')
-        return
-    elif (currency['wallet'][str(ctx.author.id)] < 5000):
-        await ctx.reply('You have less than 10k in your wallet. Don\'t be greedy.')
-        return
-    if (chance <= 20):
-        x:int = random.randint(10000, currency['wallet'][str(user.id)])
+    if currency['wallet'][str(user.id)] < 10000: return await ctx.reply('You really want to risk losing your life to a poor person? (imagine robbing someone with < 10k net worth)')
+    elif currency['wallet'][str(ctx.author.id)] < 5000: return await ctx.reply('You have less than 10k in your wallet. Don\'t be greedy.')
```
Comment (notsniped) [inline main.py:440]: Changing the value to 10000, it should not be 5000

**Comment by thatOneArchUser** [comment]
nvm merge takes insanely long

**Comment by notsniped** [comment]
I was just busy with exams, you know

**Comment by notsniped** [comment]
Ok I'm back to checking, I guess

File: main.py [hunk 9]
```diff
@@ -799,40 +598,15 @@ async def dig(ctx:SlashContext):
...
-        await ctx.reply(f'You found a {choice} while digging! Cool, right?')
-    elif (choice == "nothing"):
-        await ctx.reply('After some time of digging you eventually gave up. You got nothing.')
+        await ctx.reply(f'You found a {choice} while digging ')
+    elif (choice == "nothing"): await ctx.reply('After some time of digging you eventually gave up. You got nothing.')
     elif (choice == "died"):
         currency['wallet'][str(ctx.author.id)] -= 2000
         save()
-        await ctx.reply('YOU FELL INTO YOUR OWN TRAP AND DIED LMFAO\nYou lost 2000 coins in the process.')
+        await ctx.reply('YOU FELL INTO YOUR OWN TRAP AND DIED LMFAO\nYou lost 2000 coins in the process.') #but you were digging??
```
Comment (notsniped) [inline main.py:609]: pov: you do a neck flip falling into the hole you dug

**Comment by notsniped** [comment]
Ready to merge, just letting checks finish

**Comment by thatOneArchUser** [comment]
oih damn that was unexpcreg

**Comment by notsniped** [comment]
I went grind mode

**Comment by notsniped** [comment]
> oih damn that was unexpcreg
> 
> edit: oh* exit 2: unexpected* edit 3; edit*

edit 4: :*

**Comment by thatOneArchUser** [comment]
> > oih damn that was unexpcreg
> > edit: oh* exit 2: unexpected* edit 3; edit*
> 
> edit 4: :*

IGUDEBWNSODFIPBMPWEIOMBFPOISMDSKMGBWLFBM;DF/V

**Comment by notsniped** [comment]
L

**Pull request merged**